### ISSUE TITLE
[question] Using Artifactory Azure DevOps Extension with self-hosted VM

### ISSUE THREAD

**Issue by akalali** [conan-io/conan#6906]
I'm currently setting up Artifactory CE locally to test my Conan project to work with pre-built packages.
Until now, everything works fine, I created conanfiles for my libraries, created packages and uploaded them to my _local_ Artifactory instance, which I installed inside a Ubuntu VM (guest) running in Win10 (host).
The _local_ Artifactory instance is running on `localhost:8082` or more specifically on `http://192.168.0.171:8082` (from outside, i.e. Win10 host).
I'm using a bridged network (VirtualBox) and I'm able to add the Artifactory remote repository (inside my VM) within Win10 and I'm also able to upload and retrieve packages to and from the Artifactory instance (inside my VM) from Win10.

Next step is now to set up my Ubuntu VM as self-hosted agent for Azure to automatically build my package, upload it to the _local_ Artifactory repository and retrieve it for another library to see if everything worked well. In this case the Ubuntu VM simulates my webserver running an instance of Artifactory. Since the VM is also the agent on which the Azure pipeline is running, my first guess was to use the Artifactory Azure DevOps Extension and create an Artifactory Service for my `Artifactory Conan` task.
The _Artifactory Service_ is a service connection that requires a _Server URL_. What is this URL? I tried `http://localhost:8082` and it returned
```
The URL resolves to a loopback or localhost address. This address is not allowed for use in a Service Endpoint.
```

I tried `http://192.168.0.171:8082` and it returned
```
The URL resolves to address "http://192.168.0.171:8082/api/plugins", which is in a special purpose range that is not allowed in a Service Endpoint.
```
Obviously I don't know what is required here and if it is possible, what I'm trying to achieve here.
I thought that my self-hosted Agent should be able to upload a package to an Artifactory repository which is locally hosted by the Agent itself.
Am I missing something?
What other options do I have without setting up a real webserver that is publicly accessible via Internet?

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by eyalbe4** [comment]
@akalali,
Are you able to save the Artifactory details when defining the Artifactory service endpoint on Azure DevOps? I think that Azure DevOps triggers a connection test before allowing to save the details. Please confirm that this is what you see.
If this is the case, we can try to make this test optional in the Artifactory Azure DevOps Extension code (hopefully this is allowed in Azure DevOps).

**Comment by eyalbe4** [comment]
@akalali,
Following my previous message, you can open a feature request on https://github.com/jfrog/artifactory-azure-devops-extension/issues for allowing to disable this validation.

**Comment by akalali** [comment]
I just realized that it is possible to _save_ [the service definition] _without verification_.
I also just realized that the given address
`The URL resolves to address "http://192.168.0.171:8082/artifactory/api/plugins"`
leads to the following error 400 message:
`This REST API is available only in Artifactory Pro (see: jfrog.com/artifactory/features). If you are already running Artifactory Pro please make sure your server is activated with a valid license key.`

However, https://docs.conan.io/en/latest/uploading_packages/artifactory_ce.html and https://www.jfrog.com/confluence/display/JFROG/Artifactory+Comparison+Matrix state that it is possible to use REST API. Any comments on this?

I will try to get my example running without server-verification and update the progress here.

**Comment by czoido** [comment]
Hi @akalali,

Regarding the question about the REST API in the CE edition, I think that the problem is that you are calling the [REST API plugins endpoint](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-PLUGINS) and the user plugins feature [is not available in the CE edition](https://www.jfrog.com/confluence/display/JFROG/Artifactory+Comparison+Matrix), but for other endpoints which features are included you can call the REST API.

Please, could you please tell if you managed to make it work to see if we can close the issue?

Thanks a lot!

**Comment by akalali** [comment]
Sorry, I wasn't able to reply earlier.

I don't understand what your are saying about the endpoint. I am not calling anything, I was just setting the address of the "local" Artifactory instance inside my Ubuntu VM. What happend then was not under my control.

However, I continued with _save without verification_ and used the _Add Remote_ command to add this service instance to my conan remote list. I had to play around with the conan home and working dir etc. but finally I got it working up to the state that I am now able to add a remote and find packages inside this remote instance.

So if there is no more explanation of the plugins endpoint this issue can be closed.
